src/bot/lib/query.js

```diff
@@ -30,18 +30,19 @@ export function findSkillByName(token, search) {
 }
 
 // Find the best match for the given search term, and complain if necessary.
-export function findSkillAndHandleErrors(token, search) {
+export function findItemAndHandleErrors(token, search, queryFunction) {
```

Comment:
The name of this function itself belies a problem, the word `And` should feel like a smell. With our refactoring above, this could become

```js
// Find the best match for the given search term, and complain if necessary.
export function prepareMatchOutput(search, results) {
  const {matches, match, exact} = results;
  const output = [];
  if (matches.length === 0) {
    throw abort(`_No matches found for "${search}"._`);
  }
  else if (matches.length === 1) {
    output.push(`_You specified "${search}", which matches: *${matches[0].name}*._`);
  }
  else {
    const itemsList = matches.map(item => item.name).join(', ');
    output.push(`_Multiple matches were found: ${itemsList}._`);
    if (exact) {
      output.push(`_You specified "${search}", which matches: *${exact.name}*._`);
    }
    else {
      throw abort(`_You specified "${search}", which is ambiguous. Please be more specific._`);
    }
  }
  return {
    matches,
    match,
    exact,
    output,
  };
}
```

Now our calling code becomes:
```js
query.categoryByName({token, search})
  .then(parseMatches)
  .then(prepareMatchOutput);
```